coalib/parsing/ConfParser.py

```diff
@@ -156,7 +162,14 @@ def __parse_lines(self, lines, origin):
                                     remove_empty_iter_elements=
                                     self.__remove_empty_iter_elements),
                                     # Stop ignoring
-                            allow_appending=(keys == []))
+                            allow_appending=(
+                                keys == [] or keys == multiple_ignore_keys))
+            for key in current_section:
```

Comment:
@jayvdb You asked me, not to perform string manipulation here and rather use `Setting.__iter__`. I tried using it but is it not it's just helping me to iterate over the `Setting`, I would need to do the string manipulation even if I use it?
Further, please let me know your review on this and the test cases I have made.

Reply:
You *must* get rid of this from here.

You can do extra string manipulation in your tests only.  Test logic can be used to describe the ugly problems which havent been solved properly yet.  And add comments in the tests to explain the workarounds, and unsolved problems.